pkg/addons/device_plugin.go

```diff
@@ -234,7 +234,7 @@ func (n *NvidiaDevicePlugin) SetTolerations(spec *corev1.PodTemplateSpec) error
 	}
 	for _, ng := range n.spec.ManagedNodeGroups {
 		if api.HasInstanceTypeManaged(ng, instance.IsNvidiaInstanceType) &&
```

Comment:
It is weird to me that we even check the AMIFamily

Are there AMI families we wouldn't want to apply to this?